bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/providers/sources/AbstractSonySource.java

```diff
@@ -0,0 +1,688 @@
+
+    /** The lock used to manage listeners */
+    private final ReadWriteLock listenerLock = new ReentrantReadWriteLock();
+
+    /** The list of listeners */
+    private final Map<ServiceModelName, List<SonyModelListener>> listeners = new HashMap<>();
+
+    @Override
+    public Collection<ThingType> getThingTypes() {
+        final Lock readLock = stateLock.readLock();
+        readLock.lock();
+        try {
+            return thingTypes.values();
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public @Nullable ThingType getThingType(final ThingTypeUID thingTypeUID) {
+        Objects.requireNonNull(thingTypeUID, "thingTypeUID cannot be null");
+
+        final Lock readLock = stateLock.readLock();
+        readLock.lock();
+        try {
+            return thingTypes.get(thingTypeUID);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public @Nullable ChannelGroupType getChannelGroupType(final ChannelGroupTypeUID channelGroupTypeUID) {
+        Objects.requireNonNull(channelGroupTypeUID, "channelGroupTypeUID cannot be null");
+        final Lock readLock = stateLock.readLock();
+        readLock.lock();
+        try {
+            return groupTypes.get(channelGroupTypeUID);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public @Nullable Collection<ChannelGroupType> getChannelGroupTypes() {
+        final Lock readLock = stateLock.readLock();
+        readLock.lock();
+        try {
+            return groupTypes.values();
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public @Nullable SonyThingDefinition getSonyThingTypeDefinition(final ThingTypeUID thingTypeUID) {
+        Objects.requireNonNull(thingTypeUID, "thingTypeUID cannot be null");
+        final Lock readLock = stateLock.readLock();
+        readLock.lock();
+        try {
+            return thingTypeDefinitions.get(thingTypeUID);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Will read all files in the specified folder and store the related thing types
+     *
+     * @param folder a non-null, non-empty folder (within userdata)
+     * @throws IOException if an IO exception occurs reading the files
+     * @throws JsonSyntaxException if a json syntax error occurs
+     */
+    protected void readFiles(final String folder) throws IOException, JsonSyntaxException {
+        Validate.notEmpty(folder, "folder cannot be empty");
+
+        logger.debug("Reading all files in {}", folder);
+
+        final Lock writeLock = stateLock.writeLock();
+        writeLock.lock();
+        try {
+            // clear out prior entries
+            groupTypes.clear();
+            thingTypes.clear();
+            thingTypeDefinitions.clear();
+
+            final List<Map.Entry<ThingType, SonyThingDefinition>> results = new ArrayList<>();
```

Comment:
this only seems to be written to but never read from.

Reply:
curious - I'm pretty sure I returned it from this method at one point but for the life of me can't remember this particular code (must have been many iterations ago).  Removing!